cylc/flow/scheduler.py

```diff
@@ -909,18 +913,43 @@ def process_queued_task_messages(self) -> None:
                     should_poll = True
             if should_poll:
                 to_poll_tasks.append(itask)
-        self.task_job_mgr.poll_task_jobs(
-            self.workflow, to_poll_tasks)
+        if to_poll_tasks:
+            self.task_job_mgr.poll_task_jobs(self.workflow, to_poll_tasks)
+        # Remaining messages don't have a receiving task in the pool.
+        # E.g., after manually setting a running task to finished.
```

Comment:
Note I haven't anything here on this branch except: added the comment, and logged the ignored messages (if any).

The method retrieves all received task messages from the queue, and "processes" those that target an existing (n=0) task proxy, then polls those that cause a backward state change.

> `# Remaining messages don't have a receiving task in the pool.`

Any remaining unprocessed messages do not target an existing task proxy (otherwise they would have been forwarded to the task event manager for processing).

> `# E.g., after manually setting a running task to finished.`

That's meant to show how this could happen: if I manually set a running task to finished, the task will be removed from n=0, but the orphaned job will still return status messages.  (So yes a running task *would* have a proxy in the pool, but not *after* I `cylc set` the task to finished and orphan the job).

Hope that makes sense - should I add more detail to the comment?